source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -124,25 +124,34 @@ std::string InstanceImplWin32::fileReadToEnd(const std::string& path) {
   if (illegalPath(path)) {
     throw EnvoyException(absl::StrCat("Invalid path: ", path));
   }
-
-  std::ios::sync_with_stdio(false);
-
-  // On Windows, we need to explicitly set the file mode as binary. Otherwise,
-  // 0x1a will be treated as EOF
-  std::ifstream file(path, std::ios::binary);
-  if (file.fail()) {
+  auto fd = CreateFileA(path.c_str(), GENERIC_READ,
+                        FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, 0, OPEN_EXISTING, 0,
```

Comment:
Good call! I guess I didn't realize the implementation of `std::ifstream` doesn't already do this

Reply:
Actually there is no way to specify `FILE_SHARE_DELETE` with `std::ifstream` according to the docs [here](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/sopen-s-wsopen-s?view=msvc-160)

Reply:
🤔 huh